File: src/test/java/solutions/n1047_remove_all_adjacent_duplicates_In_string/SolutionTest.java
package solutions.n1047_remove_all_adjacent_duplicates_In_string;

import org.junit.Assert;
import org.junit.BeforeClass;
import org.junit.Test;

/**
 * @author : Sobornov Vladimir
 * @since : 10.11.2022
 **/
public class SolutionTest {

    private static Solution solution;

    @BeforeClass
    public static void beforeClass() {
        solution = new Solution();
    }

    @Test
    public void removeDuplicatesTest() {
        Assert.assertEquals("ca", solution.removeDuplicates("abbaca"));
    }

    @Test
    public void removeDuplicatesLongStringTest() {
        Assert.assertEquals("ay", solution.removeDuplicates("azxxzy"));
    }

    @Test
    public void removeDuplicatesSimpleStringTest() {
        Assert.assertEquals("a", solution.removeDuplicates("aaaaaaaaaaa"));
    }
}

File: src/test/kotlin/solutions/n852_peak_index_in_a_mountain_array/SolutionTest.kt
package solutions.n852_peak_index_in_a_mountain_array

import org.junit.Assert
import org.junit.Test

class SolutionTest {
    val solution = Solution()

    @Test
    fun peakIndexInMountainArrayTest1() {
        Assert.assertEquals(1, solution.peakIndexInMountainArray(intArrayOf(1,2,1)))
    }

    @Test
    fun peakIndexInMountainArrayTest2() {
        Assert.assertEquals(3, solution.peakIndexInMountainArray(intArrayOf(0,1,2,5)))
    }

    @Test
    fun peakIndexInMountainArrayTest3() {
        Assert.assertEquals(0, solution.peakIndexInMountainArray(intArrayOf(10,5,1)))
    }
}